HW4_Bobkov/protein_tools.py

```diff
@@ -0,0 +1,220 @@
+    return:
+    - (bool): whether pattern and fragment match
+    """
+    for i in range(0,len(sequence)):
+        if not sequence[i]==pattern[i]:
+            return False
+            break
+    return True
+    
+def find_pattern(sequences: list, pattern: str)->dict:
+    """
+    Find all non-overlaping instances of a given pattern in sequences
+    arguments:
+    - sequences (list): sequences to find the pattern in
+    - pattern (str): pattern in question
+    return
+    - finds(dict): dictionary with sequences as keys and lists of indexes of patterns and the number of patterns as values
+    """
+    finds={}
+    for j in range(0, len(sequences)):
+        find=[]
+        for i in range(0, len(sequences[j])):
+            if compare_pattern(sequences[j][i:i+len(pattern)], pattern):
+                find.append(i)
+                i+=len(pattern)
+            else:
+                continue
+        finds[sequences[j]]=[len(find)]+find
+    return finds
+
+
+def transform_to_DNA_code(protein):
+    """
+    Transforming of an amino acid sequence/protein to DNA sequence
+    :param protein: amino acid sequence of protein
+    :return: sequence of protein in the DNA sequence form 
+    """
+    retrnaslation_dict = {
+        'F': 'TTC', 'f': 'ttc',
+        'L': 'TTA', 'l': 'tta',
+        'S': 'TCG', 's': 'tcg',
+        'Y': 'TAC', 'y': 'tac',
+        'C': 'TGC', 'c': 'tgc',
+        'W': 'TGG', 'w': 'tgg',
+        'P': 'CCC', 'p': 'ccc',
+        'H': 'CAT', 'h': 'cat',
+        'Q': 'GAA', 'q': 'gaa',
+        'R': 'CGA', 'r': 'cga',
+        'I': 'ATT', 'i': 'att',
+        'M': 'ATG', 'm': 'atg',
+        'T': 'ACC', 't': 'acc',
+        'N': 'AAT', 'n': 'aat',
+        'K': 'AAA', 'k': 'aaa',
+        'V': 'GTT', 'v': 'gtt',
+        'A': 'GCA', 'a': 'gca',
+        'D': 'GAT', 'd': 'gca',
+        'E': 'GAG', 'e': 'gag',
+        'G': 'GGG', 'g': 'ggg'
+    }
+
+    return ''.join([retrnaslation_dict[i] for i in protein])
+
+
+def rename_three_letter_name (seqs: list, sep = '')->list:
+    """
+    Transform into a three-letter amino acids entry.
+    arguments:
+        - seqs (list): list of sequences for transforming to three-letter entire
+        - sep (str): separator between aminoacids, default = ''
+    return:
+        - list: transformed sequences with separators
+    """
+    res=[]
+    threel = {'A': 'ALA', 'R': 'ARG', 'N': 'ASN', 'D': "ASP", 'V': 'VAL', 
+                 'H': 'HIS', 'G': "GLY", 'Q': "GLN", 'E': 'GLU', 'I': 'ILE', 
+                 'L': 'LEU', 'K': 'LYS', 'M': 'MET', 'P': 'PRO', 'S': 'SER', 
+                 'Y': 'TYR', 'T': 'THR', 'W': 'TRP', 'F': 'PHE', 'C': 'CYS', 
+                 'a': 'ala', 'r': 'arg', 'n': 'asn', 'd': "asp", 'v': 'val', 
+                 'h': 'his', 'g': "gly", 'q': "gln", 'e': 'glu', 'i': 'ile', 
+                 'l': 'leu', 'k': 'lys', 'm': 'met', 'p': 'pro', 's': 'ser', 
+                 'y': 'tyr', 't': 'thr', 'w': 'trp', 'f': 'phe', 'c': 'cys'}
```

Comment:
В PEP8 [принято](https://peps.python.org/pep-0008/#indentation) закрывающую скобку в многострочных конструцкциях ставить на новой строке. 
Либо под первым непробельным символом:
```suggestion
                 'y': 'tyr', 't': 'thr', 'w': 'trp', 'f': 'phe', 'c': 'cys'
                 }
```
Либо по левому краю:
```suggestion
                 'y': 'tyr', 't': 'thr', 'w': 'trp', 'f': 'phe', 'c': 'cys'
}
```
Вот эти 2 варианта уже на ваше усмотрение